src/primitives/arc/mod.rs

```diff
@@ -122,9 +165,35 @@ impl PointsIter for Arc {
     }
 }
 
+// TODO Check rect.styled_bounding_box()
 impl Dimensions for Arc {
     fn bounding_box(&self) -> Rectangle {
-        Rectangle::new(self.top_left, Size::new(self.diameter, self.diameter))
+        // Create a rectangle between the start and the end angle points.
+        let rect = Rectangle::with_corners(self.point_start(), self.point_end());
+        let mut p1 = rect.top_left;
+        let mut p2 = p1 + rect.size.x_axis() + rect.size.y_axis();
```

Comment:
`p2` isn't correct, because it would be outside the rect. This should work:
```suggestion
        let start = self.point_start();
        let end = self.point_start();        
        let mut p1 = start.component_min(end);
        let mut p2 = start.component_max(end);
```